### PR TITLE
Improve jackson benchmark in java

### PR DESCRIPTION
**Pull request by komamitsu**

### CODE REVIEW

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            this.second = second;
+        }
+    }
+
+    public void run(int count, int warmupCount)
+            throws Exception
+    {
+        List<Tuple<String, double[]>> benchmarksResults = new ArrayList<Tuple<String, double[]>>(benchmarkableList.size());
+
+        for (Benchmarkable benchmark : benchmarkableList) {
```
Comment (xerial) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: The order of running programs also affects the performance results in JVM. Simply running System.gc() is not sufficient.

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            double[] durations = new double[count];
+
+            for (int i = 0; i < count + warmupCount; i++) {
+                if (i >= warmupCount) {
+                    System.gc();
```
Comment (xerial) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: And also what do you expect by running this command? There is not guarantee that JVM does something by calling gc().

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        for (Tuple<String, double[]> benchmarkResult : benchmarksResults) {
+            printStat(benchmarkResult.first, benchmarkResult.second);
+        }
+    }
+
+    private void printStat(String label, double[] values)
+    {
+        StandardDeviation standardDeviation = new StandardDeviation();
+        System.out.println(label + ":");
+        System.out.println(String.format("  mean : %8.3f", StatUtils.mean(values)));
```
Comment (xerial) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: The mean value including outliers (min, max) is not a good measure in evaluating the performance. We also need to add median or mean (average) value without containing outliers (e.g., by removing max and min, or a few percent of the top results).

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 1 above)
Comment (komamitsu) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: The benchmark has 4 times warmup runs for each one and it avoids the affect of running order, I think. BTW, how does your benchmark avoid the order of running programs?

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 2 above)
Comment (komamitsu) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: Actually, with calling the method, the variance of benchmark durations decreases. I should investigate the mechanism later.

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 3 above)
Comment (komamitsu) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: According to the result of profiling, this method accounts for 1% of the total duration. Also, we compare the performance of Jackson and MessagePack relatively with this benchmark even if this method affects a bit.

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 1 above)
Comment (xerial) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: For example, if you have two programs P1, P2,  it runs the program P1, P2, P1, P2,... to reduce the order effect.

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 3 above)
Comment (xerial) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: I just meant we should see median or average (which is not biased by too distant values).

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 1 above)
Comment (komamitsu) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: Okay. The original benchmark code (https://github.com/msgpack/msgpack-java/blob/v07-develop/msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/MessagePackDataformatPojoBenchmarkTest.java#L98-L130) does the same thing, right?

File: msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java (hunk 3 above)
Comment (komamitsu) [inline msgpack-jackson/src/test/java/org/msgpack/jackson/dataformat/benchmark/Benchmarker.java]: Sorry, I misread your comment. I'll take care of it

**Comment by komamitsu** [comment]
@xerial Can you review it again? Thanks

**Comment by xerial** [comment]
ok. It looks good.

**Pull request merged**